### DISCUSSION TITLE
Allow folk to designate alt text for cover images

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#17725] [Feature Requests]
I'd love it if we could allow authors the ability to designate alt/description text for their cover images.

We allow folks to do this on images that are placed in the body of a post using the following Markdown:

`![Image description](URL)`

If we could also extend this to cover images in some way, I feel like it'd be a nice a11y improvement!

**Comment by segdeha** [comment]
I was just about to open my own discussion when the clever “I have searched for similar discussions” prompt led me to this one! I wholeheartedly endorse this request.

Currently, the `alt` text for cover images is based on the post title. E.g., for [The History of Sticky Keys](https://dev.to/assistivlabs/the-history-of-sticky-keys-3c1d), the cover image’s `alt` text is currently `Cover image for The History of Sticky Keys`.

It would be helpful for people using assistive technologies to have more descriptive alternative text associated with cover images.

As an example, for this cover image, I would have specified the following as the `alt` text: `Screenshot of Windows 11 with the Sticky Keys modal activated. The modal asks, do you want to turn on Sticky Keys?`

<img width="1624" alt="screenshot showing the alt text for the history of sticky keys as cover image for the history of sticky keys" src="https://user-images.githubusercontent.com/4306/211917836-7815ce39-e8b9-47d0-abd5-09a6ee102ad9.png">

**Reply by segdeha** [reply]
@nickytonline has offered to partner with a developer from [The Collab Lab](https://the-collab-lab.codes) (where he’s a mentor) to implement the fix, if we can get sign off on it. Thanks!

**Reply by nickytonline** [reply]
I think what we could do here is have an alternative text field in the article model. It could default to the title if nothing is specified. If something is specified though, we'd use that instead.

In the editor this would mean adding an optional field. And when the articles render in the feed, search, etc. the article model could return a cover_alt_text field.

And in the end to end test if it's not in there yet, we could get the cover image via `cy.getByRole('img', { name: 'alt text' })`

**Reply by michael-tharrington** [reply]
Wow, thanks so much for weighing in here and rounding out this issue with more description, @segdeha!

And also thank you @nickytonline for agreeing to implement this change while partnering with someone else to make it happen. That would be so awesome!

I'll make this known to the team to help advocate and see if we can get this approved.
